src/reformat_libyuv.c

```diff
@@ -973,93 +976,87 @@ avifResult avifImageYUVToRGBLibYUV(const avifImage * image, avifRGBImage * rgb,
+            libyuvResult = lcf.yuvToRgbMatrix(image->yuvPlanes[AVIF_CHAN_Y],
+                                              image->yuvRowBytes[AVIF_CHAN_Y],
+                                              image->yuvPlanes[uPlaneIndex],
+                                              image->yuvRowBytes[uPlaneIndex],
+                                              image->yuvPlanes[vPlaneIndex],
+                                              image->yuvRowBytes[vPlaneIndex],
                                               rgb->pixels,
                                               rgb->rowBytes,
                                               matrix,
-                                              image8->width,
-                                              image8->height);
+                                              image->width,
+                                              image->height);
         } else if (lcf.yuvaToRgbMatrix != NULL) {
-            libyuvResult = lcf.yuvaToRgbMatrix(image8->yuvPlanes[AVIF_CHAN_Y],
-                                               image8->yuvRowBytes[AVIF_CHAN_Y],
-                                               image8->yuvPlanes[uPlaneIndex],
-                                               image8->yuvRowBytes[uPlaneIndex],
-                                               image8->yuvPlanes[vPlaneIndex],
-                                               image8->yuvRowBytes[vPlaneIndex],
-                                               image8->alphaPlane,
-                                               image8->alphaRowBytes,
+            libyuvResult = lcf.yuvaToRgbMatrix(image->yuvPlanes[AVIF_CHAN_Y],
+                                               image->yuvRowBytes[AVIF_CHAN_Y],
+                                               image->yuvPlanes[uPlaneIndex],
+                                               image->yuvRowBytes[uPlaneIndex],
+                                               image->yuvPlanes[vPlaneIndex],
+                                               image->yuvRowBytes[vPlaneIndex],
+                                               image->alphaPlane,
+                                               image->alphaRowBytes,
                                                rgb->pixels,
                                                rgb->rowBytes,
                                                matrix,
-                                               image8->width,
-                                               image8->height,
+                                               image->width,
+                                               image->height,
                                                /*attenuate=*/0);
             *alphaReformattedWithLibYUV = AVIF_TRUE;
         }
-        if (image8 != image) {
-            avifImageFreePlanes(&image_downshifted_to_8bit, AVIF_PLANES_ALL);
+        if (inputIsHighBitDepth) {
+            avifImageFreePlanes(&image8, AVIF_PLANES_ALL);
```

Comment:
IMPORTANT: To prevent a similar bug, add `image = NULL;` after this line. Then if new code gets added that uses `image` after this point, it will get a segmentation fault.

Reply:
Done.